### PR TITLE
dispatch-build-bottle: support self-hosted Linux and unified workflow

### PR DESCRIPTION
**Pull request by dawidd6**
- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

This is going to be needed soon, when I merge dispatch workflows into one.

Companion PR: https://github.com/Homebrew/homebrew-core/pull/80015

### CODE REVIEW

**Comment by dawidd6** [comment]
Weird typecheck errors strike again, can I get some help? @issyl0

**Comment by dawidd6** [comment]
`brew typecheck` fails:dev-cmd/dispatch-build-bottle.rb:86: Method `wheezy?` does not exist on `Homebrew::CLI::Args` https://srb.help/7003
    86 |      inputs[:wheezy] = args.wheezy?.to_s if args.wheezy?
                                                     ^^^^^^^^^^^^
  Got Homebrew::CLI::Args originating from:
    dev-cmd/dispatch-build-bottle.rb:41:
    41 |    args = dispatch_build_bottle_args.parse
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Autocorrect: Use `-a` to autocorrect
    dev-cmd/dispatch-build-bottle.rb:86: Replace with `greedy?`
    86 |      inputs[:wheezy] = args.wheezy?.to_s if args.wheezy?
                                                          ^^^^^^^

dev-cmd/dispatch-build-bottle.rb:86: Method `wheezy?` does not exist on `Homebrew::CLI::Args` https://srb.help/7003
    86 |      inputs[:wheezy] = args.wheezy?.to_s if args.wheezy?
                                ^^^^^^^^^^^^
  Got Homebrew::CLI::Args originating from:
    dev-cmd/dispatch-build-bottle.rb:41:
    41 |    args = dispatch_build_bottle_args.parse
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Autocorrect: Use `-a` to autocorrect
    dev-cmd/dispatch-build-bottle.rb:86: Replace with `greedy?`
    86 |      inputs[:wheezy] = args.wheezy?.to_s if args.wheezy?
                                     ^^^^^^^

dev-cmd/dispatch-build-bottle.rb:86: Method `wheezy?` does not exist on `Homebrew::CLI::Args` https://srb.help/7003
    86 |      inputs[:wheezy] = args.wheezy?.to_s if args.wheezy?
                                                     ^^^^^^^^^^^^
  Got Homebrew::CLI::Args originating from:
    dev-cmd/dispatch-build-bottle.rb:41:
    41 |    args = dispatch_build_bottle_args.parse
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Autocorrect: Use `-a` to autocorrect
    dev-cmd/dispatch-build-bottle.rb:86: Replace with `greedy?`
    86 |      inputs[:wheezy] = args.wheezy?.to_s if args.wheezy?
                                                          ^^^^^^^

dev-cmd/dispatch-build-bottle.rb:86: Method `wheezy?` does not exist on `Homebrew::CLI::Args` https://srb.help/7003
    86 |      inputs[:wheezy] = args.wheezy?.to_s if args.wheezy?
                                ^^^^^^^^^^^^
  Got Homebrew::CLI::Args originating from:
    dev-cmd/dispatch-build-bottle.rb:41:
    41 |    args = dispatch_build_bottle_args.parse
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Autocorrect: Use `-a` to autocorrect
    dev-cmd/dispatch-build-bottle.rb:86: Replace with `greedy?`
    86 |      inputs[:wheezy] = args.wheezy?.to_s if args.wheezy?
                                     ^^^^^^^
Errors: 4
Check https://docs.brew.sh/Typechecking for more information on how to resolve these errors.

```

```

**Comment by samford** [comment]
You'll need to add a type definition for `wheezy?` to [`cli/args/args.rbi`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cli/args.rbi) with the other `returns(T::Boolean)` methods. Basically, Sorbet can't automatically understand methods that are generated from CLI switches, so we have to manually tell it about them in an `rbi` file.

**Comment by dawidd6** [comment]
Thanks. Got help on Slack too.

**Review by MikeMcQuaid — COMMENTED** [review]
Looks good, nice work!

File: Library/Homebrew/dev-cmd/dispatch-build-bottle.rb [hunk 1]
```diff
@@ -27,8 +27,12 @@ def dispatch_build_bottle_args
              description: "Upload built bottles."
       switch "--linux",
              description: "Dispatch bottle for Linux (using GitHub runners)."
+      switch "--linux-self-hosted",
+             description: "Dispatch bottle for Linux (using self-hosted runner)."
+      switch "--wheezy",
+             description: "Use Debian Wheezy container for building the bottle on Linux."
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/dispatch-build-bottle.rb:33]: How about `--linux-wheezy` here?

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/dispatch-build-bottle.rb:33]: Or even `linux-debian-wheezy`?

Comment (dawidd6) [reply Library/Homebrew/dev-cmd/dispatch-build-bottle.rb:33]: Isn't the description enough :)?

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/dispatch-build-bottle.rb:33]: Feels a bit weird to me not not reference it in this flag but have it as `--linux-*` in others.

**Pull request merged**